lib/dry/validation/evaluator.rb

```diff
@@ -163,17 +167,26 @@ def key?
         values.key?(key_name)
       end
 
-      # Check if there are any errors under the provided path
+      # Check if there are any errors on the schema under the provided path
       #
       # @param [Symbol, String, Array] A Path-compatible spec
       #
       # @return [Boolean]
       #
       # @api public
-      def error?(path)
+      def schema_error?(path)
         result.error?(path)
       end
 
+      # Check if there are any errors on the current rule
+      #
+      # @return [Boolean]
+      #
+      # @api public
+      def rule_error?
+        key(path).opts.any?
+      end
```

Comment:
Could you add `Failures#empty?` and use it here instead of inspecting its state?

Reply:
Here it goes @solnic